third_party/amd/lib/TritonAMDGPUToLLVM/TDMUtility.cpp

```diff
@@ -0,0 +1,173 @@
+#include "TDMUtility.h"
+#include "triton/Conversion/TritonGPUToLLVM/Utility.h"
+
+namespace mlir::LLVM::AMD {
+
+namespace {
+
+// Decode a TDM descriptor from group vectors into
+// (base, [shape0, shape1], [stride0, stride1]).
+std::tuple<Value, SmallVector<Value>, SmallVector<Value>>
+decodeTDMDescriptor(RewriterBase &rewriter, Location loc,
+                    ArrayRef<Value> group0, ArrayRef<Value> group1) {
+  auto ctx = rewriter.getContext();
+  auto b = TritonLLVMOpBuilder(loc, rewriter);
+  Type globalPtrTy = ptr_ty(ctx, 1);
+
+  Value globalAddrLow = group0[2];
+  Value globalAddrHigh = b.and_(group0[3], b.i32_val(0x7FFFFFFF));
```

Comment:
As chatted internally, this is okay to get started for now. But we may want to define proper struct to avoid such i64 pack/unpack if LLVM is confused about it.